src/DotNet.Status.Web/tests/DotNet.Status.Web.Tests/TelemetryControllerTests.cs

```diff
@@ -0,0 +1,75 @@
+using Xunit;
+using DotNet.Status.Web.Controllers;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Moq;
+using Microsoft.AspNetCore.Mvc;
+using Kusto.Ingest;
+using System.Threading.Tasks;
+
+namespace DotNet.Status.Web.Tests
+{
+    public class TelemetryControllerTests
+    {
+        [Fact]
+        public void TestArcadeValidationTelemetryCollection()
+        {
+            var mockLoggerMock = new Mock<ILogger<TelemetryController>>();
+            var telemetryOptionsMock = new Mock<IOptionsSnapshot<TelemetryOptions>>();
+            var telemetryOptions = new TelemetryOptions
+            {
+                KustoIngestConnectionString = "fakekustoconnectionstring",
+                KustoDatabase = "fakekustodatbase"
+            };
+            telemetryOptionsMock.SetupGet(x => x.Value).Returns(telemetryOptions);
+            var kustoIngestResultMock = new Mock<IKustoIngestionResult>();
+            var kustoIngestClientMock = new Mock<IKustoIngestClient>();
+            kustoIngestClientMock.Setup(x => x.IngestFromStreamAsync(It.IsAny<System.IO.Stream>(), It.IsAny<KustoIngestionProperties>(), null))
+                .Returns(Task.FromResult(kustoIngestResultMock.Object));
+            var controller = new TelemetryController(mockLoggerMock.Object, telemetryOptionsMock.Object, kustoIngestClientMock.Object);
+            var result = controller.CollectArcadeValidation(new ArcadeValidationData
```

Comment:
Make this test "async Task", then "await controller.CollectionArcadeValidation".  The "Wait" below will, technically, work, but it's better to stick async all the way down. :-)

Reply:
asynced all the things :)